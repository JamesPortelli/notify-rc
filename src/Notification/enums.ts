export enum Kind {
  Success,
  Warning,
  Error,
  Info
}

export enum Position {
  TopLeft,
  TopRight,
  BottomLeft,
  BottomRight,
  Top,
  Bottom,
  Centre
}
