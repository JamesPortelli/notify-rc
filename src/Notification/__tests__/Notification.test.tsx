import React from 'react';
import { render } from '@testing-library/react';

import { Notification } from '../';
import { Kind, Position } from '../enums';

const setup = (propsOverride = {}) => {
  const props = {
    kind: Kind.Success,
    message: 'Test',
    onClose: jest.fn(),
    ...propsOverride
  };

  return {
    props,
    ...render(<Notification {...props} />)
  };
};

describe('Notification Tests', () => {
  describe('Kind', () => {
    describe('when kind is success', () => {
      it('will render success styled Notification', () => {
        const { container } = setup({ kind: Kind.Success });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when kind is warning', () => {
      it('will render warning styled Notification', () => {
        const { container } = setup({ kind: Kind.Warning });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when kind is error', () => {
      it('will render error styled Notification', () => {
        const { container } = setup({ kind: Kind.Error });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when kind is info', () => {
      it('will render info styled Notification', () => {
        const { container } = setup({ kind: Kind.Info });

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Position', () => {
    describe('when position is Top', () => {
      it('will render the notification in the Top position', () => {
        const { container } = setup({ position: Position.Top });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is Bottom', () => {
      it('will render the notification in the Bottom position', () => {
        const { container } = setup({ position: Position.Bottom });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is TopLeft', () => {
      it('will render the notification in the TopLeft position', () => {
        const { container } = setup({ position: Position.TopLeft });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is TopRight', () => {
      it('will render the notification in the TopRight position', () => {
        const { container } = setup({ position: Position.TopRight });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is BottomLeft', () => {
      it('will render the notification in the BottomLeft position', () => {
        const { container } = setup({ position: Position.BottomLeft });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is BottomRight', () => {
      it('will render the notification in the BottomRight position', () => {
        const { container } = setup({ position: Position.BottomRight });

        expect(container).toMatchSnapshot();
      });
    });

    describe('when position is Centre', () => {
      it('will render the notification in the Centre position', () => {
        const { container } = setup({ position: Position.Centre });

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Timer', () => {
    describe('when seconds are provided', () => {
      it('will render the timer', () => {
        const { queryByTestId } = setup({ seconds: 1 });

        expect(queryByTestId('timer-bar')).toBeInTheDocument();
      });
    });

    describe('when seconds are not provided', () => {
      it('will not render the timer', () => {
        const { queryByTestId } = setup();

        expect(queryByTestId('timer-bar')).not.toBeInTheDocument();
      });
    });
  });
});
