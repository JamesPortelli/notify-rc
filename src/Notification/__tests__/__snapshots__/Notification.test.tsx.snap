// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Tests Kind when kind is error will render error styled Notification 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #e41123;
  top: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="2"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Kind when kind is info will render info styled Notification 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #bbe0fd;
  top: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="3"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Kind when kind is success will render success styled Notification 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  top: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Kind when kind is warning will render warning styled Notification 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #ffb41e;
  top: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="1"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is Bottom will render the notification in the Bottom position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  bottom: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is BottomLeft will render the notification in the BottomLeft position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  bottom: 10px;
  left: 10px;
  width: 300px;
  height: 130px;
}

.c1 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is BottomRight will render the notification in the BottomRight position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  bottom: 10px;
  right: 10px;
  width: 300px;
  height: 130px;
}

.c1 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is Centre will render the notification in the Centre position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  top: 50%;
  left: 50%;
  width: 300px;
  height: 130px;
  margin-left: -160px;
  margin-top: -75px;
}

.c1 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is Top will render the notification in the Top position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  top: 0;
  left: 0;
  right: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
  float: left;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is TopLeft will render the notification in the TopLeft position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  top: 10px;
  left: 10px;
  width: 300px;
  height: 130px;
}

.c1 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`Notification Tests Position when position is TopRight will render the notification in the TopRight position 1`] = `
.c0 {
  position: fixed;
  color: white;
  background-color: #2e855f;
  top: 10px;
  right: 10px;
  width: 300px;
  height: 130px;
}

.c1 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c1 > svg {
  cursor: pointer;
}

.c2 {
  margin: 10px;
}

<div>
  <div
    class="c0"
    kind="0"
  >
    <div
      class="c1"
    >
      <svg
        data-testid="close-button"
        height="18"
        width="18"
      >
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="13"
          y2="5"
        />
        <line
          stroke="white"
          stroke-width="2"
          x1="5"
          x2="13"
          y1="5"
          y2="13"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      Test
    </div>
  </div>
</div>
`;
